### PR TITLE
INT- B-19532-approved-advance-v3

### PR DESCRIPTION
**Pull request by MInthavongsay**
## Summary

**INTEGRATION**

This PR is a v3 followup to the original [PR](https://github.com/transcom/mymove/pull/12638).  An issue was discovered during PO demo where a PPM is in REJECT status and advance number is modified and SAVED, the status is automatically displayed as ACCEPT when PPM view is reloaded after SAVED. This means status of REJECTED was overwritten to EDITED/ACCEPT. After consulting with Dan Matthews, it was decided the UI shall programmatically select ACCEPT if user enters a new advance number regardless if REJECT is selected. It is assumed user's intent is to ACCEPT if he/she modifies the advance request number.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Steps to test: 

1. Log in as Services Counselor in Office and edit shipment on PPM

2. Enter an amount for Amount Requested and select Approve and hit Save & Continue

3. Edit Shipment again and edit the Amount Requested to different number but take note of saved value. Click REJECT radio button. Note the Advance Requested input reverts back to the persisted value.

4. Now, change advance number again, the status should automatically change to ACCEPT from REJECT. Modifying request in REJECT assumes user wants to ACCEPT before saving.

_Given I am a SC on the "Edit shipment details" page,
When I am reviewing the "Incentive & advance" area of the page,
Then when I update the advance amount,  Accept radio button is automatically selected if  Reject was previously selected._

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### CODE REVIEW

**Review by antgmann — APPROVED** [review]
image

Works like a charm. Code looks good

**Comment by ajlusk** [comment]
I was able to **change the value and select reject** by selecting reject and the pasting a new value into the amount requested field. The API returns that it took the new value and changes the advanceStatus to EDITED which shows as Approved in the UI.

File: src/components/Office/ShipmentIncentiveAdvance/ShipmentIncentiveAdvance.jsx [hunk 1]
```diff
@@ -95,6 +109,7 @@ const ShipmentIncentiveAdvance = ({ estimatedIncentive, advanceAmountRequested }
                     thousandsSeparator=","
                     lazy={false} // immediate masking evaluation
                     prefix="$"
+                    onChange={handleAdvanceChange}
```
Comment (ajlusk) [inline src/components/Office/ShipmentIncentiveAdvance/ShipmentIncentiveAdvance.jsx:112]: Adding `onPaste={handleAdvanceChange}` alongside onChange prevents the issue described in my previous comment but ends up with both the initial and pasted values in the the field.

Comment (ajlusk) [reply src/components/Office/ShipmentIncentiveAdvance/ShipmentIncentiveAdvance.jsx:112]: Looking through some old commits it looks like `onBlur` might be what is supposed to be used with `<MaskedTextField>`.

**Review by ajlusk — APPROVED** [review]
Unable to break it with the latest changes. Looks good.

In hindsight, it might have made more sense to just reset and disable the amount field if reject is selected rather than try to have them affect each other.

**Pull request merged**